Xamarin.Essentials/Contacts/Contacts.ios.cs

```diff
@@ -0,0 +1,106 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Contacts;
+using ContactsUI;
+
+namespace Xamarin.Essentials
+{
+    public static partial class Contacts
+    {
+        static Task<Contact> PlatformPickContactAsync()
+        {
+            var uiView = Platform.GetCurrentViewController();
+            if (uiView == null)
+                throw new ArgumentNullException($"The View Controller can't be null.");
+
+            var source = new TaskCompletionSource<Contact>();
+
+            using var picker = new CNContactPickerViewController
+            {
+                Delegate = new ContactPickerDelegate(phoneContact =>
+                    source?.TrySetResult(Contacts.GetContact(phoneContact)))
+            };
+
+            uiView.PresentViewController(picker, true, null);
+
+            return source.Task;
+        }
+
+        internal static Contact GetContact(CNContact contact)
+        {
+            if (contact == null)
+                return default;
+
+            try
+            {
+                var contactType = ToPhoneContact(contact.ContactType);
+                var phones = new List<ContactPhone>();
+
+                foreach (var item in contact.PhoneNumbers)
+                    phones.Add(new ContactPhone(item?.Value?.StringValue, contactType));
+
+                var emails = new List<ContactEmail>();
+
+                foreach (var item in contact.EmailAddresses)
+                    emails.Add(new ContactEmail(item?.Value?.ToString(), contactType));
+
+                var name = string.Empty;
+
+                if (string.IsNullOrEmpty(contact.MiddleName))
+                    name = $"{contact.GivenName} {contact.FamilyName}";
+                else
+                    name = $"{contact.GivenName} {contact.MiddleName} {contact.FamilyName}";
```

Comment:
Not so happy with this... Is it possible to have both the "display name" and each name part? Not sure how Android stores the parts, but is it possible to get it?

@Redth @pictos I am more thinking that the app displays "Last, First" or something. I think having both will add more value.

Reply:
Yeah I think display name is more about how you have setup the contacts app on the phone to display the names, and it isn't _really_ appropriate to use that here.  I think this is fine how it is, although part of me wonders if we should break up the Name for the contact into First/Middle/Last properties...

Reply:
That might be nice. They may want to show initials in a circle as well.

Reply:
Leaving as single Name for now.  We can always add First/Last later if we want, and still keep Name as a combined full name field without breaking API.